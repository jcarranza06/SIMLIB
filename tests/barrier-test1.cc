// Barrier: simple test of basic functionality
#include <simlib.h>

Barrier b("b", 5);

class BarrierTest1 : public Process {
    void Behavior(void) {
        Print("b:start %s\n", Name().c_str());
        b.Wait(); // wait on barrier
        Print("b:end %s\n", Name().c_str());
    }
};

int main()
{
    //DebugON();
    Init(0, 5);
    for (unsigned a=0; a < b.Height()+2; ++a)
        (new BarrierTest1)->Activate();
    Run();
    b.Output();
}
