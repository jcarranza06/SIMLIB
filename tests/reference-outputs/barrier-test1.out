b:start Process#0
b:start Process#1
b:start Process#2
b:start Process#3
b:start Process#4
b:start Process#5
b:start Process#6
b:end Process#0
b:end Process#1
b:end Process#2
b:end Process#3
b:end Process#4
Barrier: b
  0: Process#5
  1: Process#6
  2: empty
  3: empty
  4: empty

